soc/arm/nxp_imx/rt5xx/power.c

```diff
@@ -0,0 +1,223 @@
+				       flexspi_pins_default_state->pin_cnt,
+				       PINCTRL_REG_NONE);
+	}
+#endif
+
+#if DT_NODE_HAS_STATUS(FLEXSPI2_DEVICE, okay)
+	/* Apply flexspi2 pinctrl state for default state */
+	if (flexspi2_pin_reconfig) {
+		pinctrl_configure_pins(flexspi2_pins_default_state->pins,
+				       flexspi2_pins_default_state->pin_cnt,
+				       PINCTRL_REG_NONE);
+	}
+#endif
+}
+
+/* Invoke Low Power/System Off specific Tasks */
+__weak void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	/* FIXME: When this function is entered the Kernel has disabled
+	 * interrupts using BASEPRI register. This is incorrect as it prevents
+	 * waking up from any interrupt which priority is not 0. Work around the
+	 * issue and disable interrupts using PRIMASK register as recommended
+	 * by ARM.
+	 */
+
+	/* Set PRIMASK */
+	__disable_irq();
+
+	/* Set BASEPRI to 0 */
+	irq_unlock(0);
+
+	switch (state) {
+	case PM_STATE_RUNTIME_IDLE:
+		POWER_EnterSleep();
+		break;
+	case PM_STATE_SUSPEND_TO_IDLE:
+		set_deepsleep_pin_config();
+		POWER_EnterDeepSleep(EXCLUDE_FROM_DEEPSLEEP);
+		restore_deepsleep_pin_config();
+		break;
+	case PM_STATE_SOFT_OFF:
+		set_deepsleep_pin_config();
+		POWER_EnterDeepPowerDown(EXCLUDE_FROM_DEEP_POWERDOWN);
+		break;
+	default:
+		LOG_DBG("Unsupported power state %u", state);
+		break;
+	}
+}
+
+/* Handle SOC specific activity after Low Power Mode Exit */
+__weak void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(state);
+	ARG_UNUSED(substate_id);
+
+	/* Clear PRIMASK */
+	__enable_irq();
+}
+
+/* Initialize power system */
+static int rt5xx_power_init(const struct device *dev)
+{
+	int ret = 0;
+
+#if DT_NODE_HAS_STATUS(FLEXSPI_DEVICE, okay)
+	flexspi_pin_reconfig = false;
+
+	ret = pinctrl_lookup_state(flexspi_pincfg,
+				   PINCTRL_STATE_DEFAULT,
+				   &flexspi_pins_default_state);
+
+	if (ret == 0) {
+		/*
+		 * Default flexspi pin configuration is available, check if flexspi pin
+		 * configuration for sleep state is available.
+		 */
+		ret = pinctrl_lookup_state(flexspi_pincfg,
+					   PINCTRL_STATE_SLEEP,
+					   &flexspi_pins_sleep_state);
+		if (ret == 0) {
+			/* Flexspi pin configuation is available */
+			flexspi_pin_reconfig = true;
+		}
+	}
+#endif
+
+#if DT_NODE_HAS_STATUS(FLEXSPI2_DEVICE, okay)
+	flexspi2_pin_reconfig = false;
+
+	ret = pinctrl_lookup_state(flexspi2_pincfg,
+				   PINCTRL_STATE_DEFAULT,
+				   &flexspi2_pins_default_state);
+
+	if (ret == 0) {
+		/*
+		 * Default flexspi2 pin configuration is available, check if flexspi2 pin
+		 * configuration for sleep state is available.
+		 */
+		ret = pinctrl_lookup_state(flexspi2_pincfg,
+					   PINCTRL_STATE_SLEEP,
+					   &flexspi2_pins_sleep_state);
+		if (ret == 0) {
+			/* Flexspi pin configuation is available */
+			flexspi2_pin_reconfig = true;
+		}
+	}
+#endif
+
+	POWER_SetLdoVoltageForFreq(SystemCoreClock, 0);
```

Comment:
This API configures the RT500's internal LDO, not used with the PMIC.  Instead call BOARD_SetPmicVoltageForFreq()

Reply:
I have added a comment on why this API is called.